R/cdisc_data.R

```diff
@@ -68,58 +68,80 @@ cdisc_data <- function(...,
+  ) {
+
+    update_join_keys_to_primary(data_objects, join_keys)
 
-  retrieve_parents <- function(x) {
-    tryCatch(
-      x$get_parent(),
-      error = function(cond) rep(character(0), length(x$get_datanames()))
+    retrieve_parents <- function(x) {lifecycle::deprecate_warn(
+      when = "0.3.1",
+      "cdisc_data(
+        data_objects = 'should use data directly. Using TealDatasetConnector and TealDataset is deprecated.'
+      )"
     )
-  }
+      tryCatch(
+        x$get_parent(),
+        error = function(cond) rep(character(0), length(x$get_datanames()))
+      )
+    }
 
-  new_parents_fun <- function(data_objects) {
-    lapply(data_objects, function(x) {
+    new_parents_fun <- function(data_objects) {
+      lapply(data_objects, function(x) {
+        if (inherits(x, "TealDataConnector")) {
+          unlist(new_parents_fun(x$get_items()), recursive = FALSE)
+        } else {
+          list(retrieve_parents(x))
+        }
+      })
+    }
+
+    new_parents <- unlist(new_parents_fun(data_objects), recursive = FALSE)
+
+    names(new_parents) <- unlist(lapply(data_objects, function(x) {
       if (inherits(x, "TealDataConnector")) {
-        unlist(new_parents_fun(x$get_items()), recursive = FALSE)
+        lapply(x$get_items(), function(z) z$get_dataname())
       } else {
-        list(retrieve_parents(x))
+        x$get_datanames()
       }
-    })
-  }
-
-  new_parents <- unlist(new_parents_fun(data_objects), recursive = FALSE)
+    }))
 
-  names(new_parents) <- unlist(lapply(data_objects, function(x) {
-    if (inherits(x, "TealDataConnector")) {
-      lapply(x$get_items(), function(z) z$get_dataname())
-    } else {
-      x$get_datanames()
+    if (is_dag(new_parents)) {
+      stop("Cycle detected in a parent and child dataset graph.")
     }
-  }))
+    join_keys$set_parents(new_parents)
+    join_keys$update_keys_given_parents()
 
-  if (is_dag(new_parents)) {
-    stop("Cycle detected in a parent and child dataset graph.")
-  }
-  join_keys$set_parents(new_parents)
-  join_keys$update_keys_given_parents()
+    x <- TealData$new(..., check = check, join_keys = join_keys)
 
-  x <- TealData$new(..., check = check, join_keys = join_keys)
+    if (length(code) > 0 && !identical(code, "")) {
+      x$set_pull_code(code = code)
+    }
 
-  if (length(code) > 0 && !identical(code, "")) {
-    x$set_pull_code(code = code)
-  }
+    x$check_reproducibility()
+    x$check_metadata()
 
-  x$check_reproducibility()
-  x$check_metadata()
-  return(x)
+    if (is_pulled(x)) {
+      new_tdata(
+        env = lapply(x$get_datasets(), function(x) x$get_raw_data()),
+        code = x$get_code(),
+        keys = x$get_join_keys()
+      )
+    } else {
+      x
+    }
+  } else {
+    if (!checkmate::test_names(names(data_objects), type = "named")) {
+      stop("Dot (`...`) arguments on `teal_data()` must be named.")
+    }
```

Comment:
```suggestion
    checkmate::assert_names(names(data_objects), .var.name = "dot arguments (`...`) in `teal_data()`")
```

Reply:
Ah great!

The function I used before is the soon-to-be-deprecated `assert_named()` which had a non-descriptive error message (`failed: Must have Object.`), but this is much cleaner :-)